### PR TITLE
Database

### PR DESCRIPTION
**Pull request by markus2330**

### CODE REVIEW

File: doc/architecture/context.md [hunk 1]
```diff
@@ -58,20 +58,32 @@ Global value and local values in the Soil layer:
 
 ## Plant Database
 
-- photos
-- genus and species
-- other general infos
-- size (height, width, depth)
+- hierarchy: family, subfamily (MISSING), genus, species, variety
+- pointer to one hierarchy up (MISSING)
```
Comment (markus2330) [inline doc/architecture/context.md:62]: This is maybe not needed (depending on design)

File: doc/architecture/context.md [hunk 2]
```diff
@@ -58,20 +58,32 @@ Global value and local values in the Soil layer:
...
-  - recommended distances to other plants
+  - good/bad neighbors to other specific plants of any hierarchy (MISSING)
+  - recommended distances to any other plant, needed for area calculations (MISSING)
   - sun
   - soil weight
   - pH value of topsoil
-  - yield grade
-  - freeze resistance
-  - good/bad neighbors
+  - nutrition demand (MISSING, start with "Nutritionally poor soil" in "Environmental Tolerances")
```
Comment (markus2330) [inline doc/architecture/context.md:79]: It should be an ENUM: light feeder, moderate feeder, heavy feeder

I just saw that "Nutritionally poor soil" are only 4 plants, so not really worth to program this.

File: doc/architecture/context.md [hunk 3]
```diff
@@ -58,20 +58,32 @@ Global value and local values in the Soil layer:
...
+  - freeze resistance (hardiness zone)
   - wet/dry
-  - preferable permaculture zone (needs visits rare or often)
+  - preferable permaculture zone (MISSING)
```
Comment (markus2330) [inline doc/architecture/context.md:82]: should be smallint -1..6 (-1 should be printed as 00)

**Review by markus2330 — COMMENTED** [review]
A few comments for @aidnurs

File: doc/architecture/context.md (hunk 3 above)
Comment (aidnurs) [reply doc/architecture/context.md:82]: if we proceed with smallint, custom printing should be done on the application level. I will add this note to the documentation

**Comment by aidnurs** [comment]
Thank you for the notes @markus2330! Mixing the implementation of 2 separate PRs in a single one was not the best choice of mine(referring #42). Let's close this PR since I'm already losing the overview over all the database details. I've taken all your changes over and implemented your suggestions. Feel free to leave the notes either directly in the documentation or in the comments section of the PR #42

**Pull request closed without merging**